nano/core_test/processing_queue.cpp

```diff
@@ -30,6 +30,8 @@ TEST (processing_queue, process_one)
 	ASSERT_TIMELY (5s, processed == 1);
 	ASSERT_ALWAYS (1s, processed == 1);
 	ASSERT_EQ (queue.size (), 0);
+
+	queue.stop ();
```

Comment:
I believe these will cause a segfault if the test fails. Since the call to stop is no longer associated with a destructor.

Reply:
True, if it fails it will be messy. I added a RAII wrapper that ensures stop will always be called